### PR TITLE
Data processing API

### PR DESCRIPTION
**Pull request by lymereJ**
This pull request implements the data processing API.

### CODE REVIEW

File: src/api/data_processing.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        Args:
+            data (str): Path to the data (CSV format) to be loaded for processing.
+            data_source (str): Data source name. Use `EnergyPlus` or `Other`.
+            timestamp_column_name (str): Name of the column header that contains the time series timestamps.
+        """
+        if data_path == None:
+            logging.error("A `data_path` argument should be provided.")
+            if data_source == None:
+                logging.error("A `data_source` argument should be provided.")
+            return None
```
Comment (leijerry888) [inline src/api/data_processing.py]: should this be separate checks (not nested) to account for the case of data_path is valid and data_source is None?

Comment (lymereJ) [reply src/api/data_processing.py]: That's correct. Fixed! Thanks!

File: src/api/data_processing.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                        self.data = self.data[start_time:end_time]
+                    else:
+                        return self.data[start_time:end_time]
+            else:
+                logging.error("The end_time argument is not a Python datetime object.")
+        else:
+            logging.error("The start_time argument is not a Python datetime object.")
+        return None
+
+    def add_parameter(self, name: str = None, value: float = None, inplace=False):
```
Comment (leijerry888) [inline src/api/data_processing.py]:
```suggestion
    def add_parameter(self, name: str = None, value: float = None, inplace=False) -> None | pd.DataFrame : 
```

**Comment by lymereJ** [comment]
@leijerry888, @yanchenpnnl, @yunjoonjung-PNNL - I think this first draft is ready for review. All the items for this API have been implemented except `check_for_verif()` which needs to have #54 be merged first. Please note that Black was recently updated and some of the "base" files had to be updated so the tests could run, that is why the PR shows that a lot of files have been modified). Perhaps we need another PR for these formatting changes?

**Comment by leijerry888** [comment]
> @leijerry888, @yanchenpnnl, @yunjoonjung-PNNL - I think this first draft is ready for review. All the items for this API have been implemented except `check_for_verif()` which needs to have #54 be merged first. Please note that Black was recently updated and some of the "base" files had to be updated so the tests could run, that is why the PR shows that a lot of files have been modified). Perhaps we need another PR for these formatting changes?

Thanks Jeremy. It seems like the black caused changes are manageable in this PR. I think we don't need to do another PR for formatting

**Review by leijerry888 — APPROVED** [review]
@lymereJ Thanks for putting this together! I finished my review and committed my changes to your code. I did not add github inline comments with my changes but I did (tried) organizing my changes by commits so hopefully it can be somewhat helpful. Please let me know if anything seems to be off.

**Comment by lymereJ** [comment]
@leijerry888 - It all looks great! Thanks!

File: src/api/data_processing.py [hunk 3]
```diff
@@ -0,0 +1,593 @@
...
+
+        Args:
+            data (str): Path to the data (CSV format) to be loaded for processing.
+            data_source (str): Data source name. Use `EnergyPlus` or `Other`.
+            timestamp_column_name (str): Name of the column header that contains the time series timestamps.
+        """
+
+        self.data = None
+
+        if data_path == None:
```
Comment (lymereJ) [inline src/api/data_processing.py]: @leijerry888 - I actually have one comment. It looks like we currently have inconsistent if statement comparing a variable value to `None`. Should we convert them all to `is None`? [Here](https://stackoverflow.com/questions/3257919/what-is-the-difference-between-is-none-and-none)'s a reference.

Comment (leijerry888) [reply src/api/data_processing.py]: @lymereJ Good point. Done.

**Pull request merged**